consensus/state.go

```diff
@@ -571,8 +571,11 @@ func (cs *ConsensusState) receiveRoutine(maxSteps int) {
 		var mi msgInfo
 
 		select {
-		case height := <-cs.mempool.TxsAvailable():
-			cs.handleTxsAvailable(height)
+		case txAvailable := <-cs.mempool.TxsAvailable():
+			// TODO why don't we have a "default" branch for this drain? Similar to the publish, wouldn't this lock otherwise?
```

Comment:
this won't lock because we have other cases below

Reply:
Ah, OK, understood.